### ISSUE TITLE
Use happyzoom for minzoom, if applicable

### ISSUE THREAD

**Issue by GretaCB** [mapbox/mapnik-omnivore#120]
If a TIFF file is [a happytiff](https://github.com/mapbox/node-happytiff), no need to include semi-empty tiles for zoom levels below the happytiff's preferred zoom level (aka "happy zoom").

**If mapnik-omnivore detects a happy tiff, minzoom = z property returned from node-happytiff.**

- [x] Add a check at the end of [the getZooms function](https://github.com/mapbox/mapnik-omnivore/blob/master/lib/raster.js#L199) that runs the TIFF through node-happytiff.

If it's a happytiff:
- [ ] If the [minzoom calculated by mapnik-omnivore](https://github.com/mapbox/mapnik-omnivore/blob/master/lib/raster.js#L215) is less than `z` property from node-happytiff, return the happyzoom

If it's not a happytiff:
- [x] Return [minzoom calculated by mapnik-omnivore](https://github.com/mapbox/mapnik-omnivore/blob/master/lib/raster.js#L215) 
- [x] If a tiff is happy, node-happytiff will return a `z` property. If no `z` property is returned, it's a sadtiff

node-happytiff returns JSON that looks like:
```
{
  "z": 7,
  "x": 91,
  "y": 50
}
```

node-tiff returns an error for sadtiffs.
```
Error: Extent coord 0 differs from nearest tile (12/792/1480) by more than > 0.000343 (-110.37655746458053 vs -110.390625)
```

- [x] Add tests

cc @camillacaros

**Comment by rclark** [comment]
Do we want the conditions, or would it be more clear to just say "if it is a happytiff, that's what our minzoom will be"? What is the purpose of _ever_ using the omnivore minzoom in the case of a happytiff?

**Comment by GretaCB** [comment]
Omnivore's minzoom is kind of made up, so not sure why we wouldn't prefer the happytiff's zoom every time.

I can't think of any scenario where we would want semi-complete tiles (1/4 data and 3/4 nodata)

**Comment by camillacaros** [comment]
Above plan is updated [per conversation](https://mapbox.slack.com/archives/satellite/p1438636958000376) about necessary/unecessary checks. We will now be looking at whether a tiff is a happy tiff or not. If YES, use happytiff minzoom as minzoom in mapnik-omnivore.

**Comment by camillacaros** [comment]
per closed PR #121 this issue will need to be addressed in a different toolset. blocked here by circular deps.
